### PR TITLE
Add violin plots and some other minor tweaks

### PR DESCRIPTION
**Pull request by jsundram**

### CODE REVIEW

**Comment by jsundram** [comment]
I added a few violin plots to show the distribution of times as well.
[
Solve Times as of 2021-06-29
Solve Times Split Violin Plot as of 2021-06-29
Solve Times Violin Plot as of 2021-06-29
](url)

File: plot/plot.py [hunk 1]
```diff
@@ -63,38 +67,83 @@ def parse_data(csv_path):
 
 
 def save_plot(df, out_path):
-    DAYS = df["weekday"].unique()
-
     # Pick an appropriate y-axis, balancing being robust to outliers vs. showing all data
     Y_MAX = df["solve_time_secs"].quantile(0.99) / 60
+    Y_MAX = 60 # unfortunately, an hour is a good upper limit for me.
```
Comment (kesyog) [inline plot/plot.py]: Can you make this an optional argument that can override the existing auto-ranging behavior?

File: plot/plot.py [hunk 2]
```diff
@@ -63,38 +67,83 @@ def parse_data(csv_path):
...
+    # https://seaborn.pydata.org/examples/kde_ridgeplot.html
+    df['solve_time_m'] = df['solve_time_secs'] / 60.0
+    ax = sns.violinplot(x="weekday", y="solve_time_m", data=df, order=DAYS)
+
+    date = max(df['Solved datetime']).strftime("%b %d, %Y")
+    ax.set_title("%d NYT Crossword Solve Times by Day of Week as of %s" % (len(df), date))
+    ax.set_xlabel("Day of Week")
+    ax.set_ylabel("Minutes to Solve")
+
+    ax.set_ylim(0, 65)
```
Comment (kesyog) [inline plot/plot.py]: It'd be nice if the y axis limit weren't hard-coded for the two violin plots as well. I'd be fine with creating an issue to tackle this later as well.

**Review by kesyog — CHANGES_REQUESTED** [review]
Hey, thanks for the PR! The violin plots and updated style are great additions. I just had a couple minor comments about the hard-coded y-axis limits, but it otherwise looks good to me.

File: plot/plot.py (hunk 1 above)
Comment (jsundram) [reply plot/plot.py]: done, good idea!

File: plot/plot.py (hunk 2 above)
Comment (jsundram) [reply plot/plot.py]: done, good idea!

**Review by kesyog — APPROVED** [review]

**Pull request merged**